doc/source/dev/core-dev/deprecations.rst.inc

```diff
@@ -7,25 +7,41 @@ There are various reasons for wanting to remove existing functionality: it's
 buggy, the API isn't understandable, it's superseded by functionality with
 better performance, it needs to be moved to another SciPy submodule, etc.
 
-In general it's not a good idea to remove something without warning users about
-that removal first.  Therefore this is what should be done before removing
+In general, it's not a good idea to remove something or make other backwards
+compatibility breaking changes without warning users about
+the change first. Therefore this is what should be done before deprecating
 something from the public API:
 
 #. Propose to deprecate the functionality on the scipy-dev mailing list and get
    agreement that that's OK.
-#. Add a ``DeprecationWarning`` for it, which states that the functionality was
-   deprecated, and in which release. For Cython APIs, see
+#. Add a ``DeprecationWarning`` that is triggered by using the deprecated API.
+   The ``stacklevel`` must be set correctly so that the warning refers
+   to the function’s caller, rather than the ``warnings.warn()`` call itself.
+   For Cython APIs, see
    :ref:`deprecating-public-cython-api` for the practical steps.
-#. Mention the deprecation in the release notes for that release.
-#. Wait till at least 6 months after the release date of the release that
+#. Use the ``.. deprecated::`` directive to document the deprecation. The rest
+   of the documentation should be left until the function is removed and the
+   function should be left in the toctree. Deprecation warnings causing the doc
+   build to fail can be resolved by adding an entry to the
+   `conf.py <https://github.com/scipy/scipy/blob/main/doc/source/conf.py#L318>`_
+   file.
+#. Ensure that both these warning messages contain the following information:
+
+   #. What has been deprecated?
+   #. What version it was deprecated in?
+   #. What version the deprecation will be executed?
+   #. What the user should use instead?
```

Comment:
```suggestion
   #. What exactly has been deprecated?
   #. What should the user do instead (resp. what does the user need to do to silence the warning)?
   #. In which SciPy version will the deprecation be executed?
```

I think the "when was it deprecated" doesn't _need_ to be in the DeprecationWarning. This is even moreso the case where deprecations happen in stages, e.g. first in documentation and then with the actual warning - which version would be accurate then? I think it's not relevant information, the user already knows which version they are on, relevant is when things will break and how to avoid the warning.